backend/src/openarchiefbeheer/zaken/api/serializers.py

```diff
@@ -86,3 +86,21 @@ class SelectielijstklasseChoicesQueryParamSerializer(serializers.Serializer):
             "The URL of the zaak for which the selectielijstklasse choices are needed."
         )
     )
+
+
+class ZaakTypeChoicesQueryParamSerializer(serializers.Serializer):
+    destruction_list = serializers.UUIDField(
+        help_text=_("The UUID of the destruction list to get Zaaktype choices for."),
+        required=False,
+    )
+
+    def to_internal_value(self, data: dict) -> dict:
+        from openarchiefbeheer.destruction.models import DestructionList
+
+        internal_value = super().to_internal_value(data)
+
+        if destruction_list := internal_value.get("destruction_list"):
+            internal_value["destruction_list"] = DestructionList.objects.get(
+                uuid=destruction_list
+            )
+        return internal_value
```

Comment:
This feels a bit like a hack? :thinking: 

Maybe a `validate` that checks that the destruction list with that UUID exists, but then leave getting the destruction list to the view?